src/ResourceManager/Websites/Commands.Websites/Cmdlets/BackupRestore/RestoreAzureDeletedWebApp.cs

```diff
@@ -0,0 +1,105 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.WebApps.Cmdlets.WebApps;
+using Microsoft.Azure.Commands.WebApps.Models;
+using Microsoft.Azure.Management.WebSites.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.BackupRestore
+{
+    /// <summary>
+    /// Restores a deleted Azure Web App's contents and settings to an existing web app
+    /// </summary>
+    [Cmdlet("Restore", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DeletedWebApp"), OutputType(typeof(void))]
+    public class RestoreAzureDeletedWebApp : WebAppBaseClientCmdLet
+    {
+        // Note, there can be multiple deleted web apps with the same name and resource group. Internally each deleted app has a unique ID.
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = "The deleted Azure Web App.", ValueFromPipeline = true)]
+        public AzureDeletedWebApp InputObject;
+
+        [Parameter(Position = 1, Mandatory = true, HelpMessage = "The resource group containing the new Azure Web App.", ValueFromPipelineByPropertyName = true)]
+        public string TargetResourceGroupName { get; set; }
```

Comment:
In the design review this was not mandatory - why has this changed?  Also, none of these parameters should have ValueFromPipelineByPropertyName = true, please remove from all.

Reply:
Changed this to be optional